DataProcessing/datax-host/src/main/scala/datax/processor/CommonProcessorFactory.scala

```diff
@@ -608,8 +618,18 @@ object CommonProcessorFactory {
           val blobStorageKey = ExecutorHelper.createBlobStorageKeyBroadcastVariable(paths.head, spark)
 
           val inputDf = internalFiles
-            .flatMap(file => HadoopClient.readHdfsFile(file.inputPath, gzip = file.inputPath.endsWith(".gz"), blobStorageKey)
-              .filter(l => l != null && !l.isEmpty).map((file, outputPartitionTime, _)))
+            .flatMap(file => {
+              val timeLast = System.nanoTime()
+              val retVal = HadoopClient.readHdfsFile(file.inputPath, gzip = file.inputPath.endsWith(".gz"), blobStorageKey)
+                .filter(l => l != null && !l.isEmpty).map((file, outputPartitionTime, _))
+              val timeNow = System.nanoTime()
+              AppInsightLogger.trackEvent(
+                ProductConstant.ProductRoot + "/ReadBlob",
+                Map("Timestamp" -> timeNow.toString, "InputPath" -> file.inputPath),
```

Comment:
can we get the timetaken in a single value instead of logging start and stop? seems it would help metrics building.

Reply:
Oh, this is a single metric that executes after the read operation and tracks the timediff before and after the operation